[BUG] swift_package_build/test swallow compiler diagnostics on failure

## Description

When `swift_package_build` or `swift_package_test` fails, the error response contains only `"Error: Swift package build failed"` with zero compiler diagnostics — no file path, no line number, no error message. This makes the tools unusable for iterative development where an agent needs to read and fix compile errors.

The `build_sim` tool (xcodebuild path) correctly returns file:line:col + error messages on failure, so this is specific to the swift-package workflow.

## Reproduction

1. Create a deliberate compile error in any Swift package (e.g., `let broken: DOESNOTEXIST = 42`)
2. Call `swift_package_build` with the package path
3. Response: `Error: Swift package build failed` — nothing else

Running `swift build` directly on the same package produces full diagnostics:
\`\`\`
WithLock.swift:10:25: error: cannot find type 'DOESNOTEXIST' in scope
let broken: DOESNOTEXIST = 42
                         ^
\`\`\`

## Affected tools

- `swift_package_build` (MCP and CLI)
- `swift_package_test` (MCP and CLI — same code pattern)

Both MCP and CLI modes exhibit the same behavior (`npx xcodebuildmcp swift-package build` and `--output json`), confirming the bug is in the shared response-building layer.

## Root cause (suspected)

Swift Package Manager writes compiler diagnostics to **stdout**, not stderr. The error-handling code appears to prefer `result.error` (stderr) over `result.output` (stdout) when constructing the failure response. Since stderr is an empty string `""` on SPM failures, and `""` is not nullish, the actual diagnostics in stdout are never included.

Pseudocode of the likely issue:
\`\`\`typescript
// Current (broken): ?? only skips null/undefined, not empty string
const errorMessage = result.error ?? result.output ?? 'Unknown error';

// Fix: || treats "" as falsy and falls through to result.output
const errorMessage = result.error || result.output || 'Unknown error';
\`\`\`

This pattern would exist in both `swift_package_build.ts` and `swift_package_test.ts`.

## Environment

- XcodeBuildMCP: v2.1.0 (via `npx xcodebuildmcp@latest`)
- Xcode: 26.2
- macOS: 26.3
- Node: v24.5.0
- npm: 11.5.1

## Related

- #177 — similar class of issue where test output is not returned to the agent

## Doctor Output

```
- Server Version: 2.1.0
- Xcode: 26.2 (Build 17C52)
- Node: v24.5.0
- Platform: darwin arm64, Apple M4 Max, 128GB
- AXe: v1.4.0
- DAP: lldb-dap available
- 79 tools across 15 workflows
- Xcode IDE Bridge: Not connected (mcpbridge not found)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Changelog
 
+## [Unreleased]
+
+### Fixed
+
+- Fixed `swift_package_build`, `swift_package_test`, and `swift_package_clean` swallowing compiler diagnostics on failure by treating empty stderr as falsy, so stdout diagnostics are included in the error response ([#243](https://github.com/getsentry/XcodeBuildMCP/issues/243)).
+
 ## [2.1.0]
 
 ### Added

--- a/src/mcp/tools/swift-package/__tests__/swift_package_build.test.ts
+++ b/src/mcp/tools/swift-package/__tests__/swift_package_build.test.ts
@@ -228,6 +228,32 @@ describe('swift_package_build plugin', () => {
       });
     });
 
+    it('should include stdout diagnostics when stderr is empty on build failure', async () => {
+      const executor = createMockExecutor({
+        success: false,
+        error: '',
+        output:
+          "main.swift:10:25: error: cannot find type 'DOESNOTEXIST' in scope\nlet broken: DOESNOTEXIST = 42",
+      });
+
+      const result = await swift_package_buildLogic(
+        {
+          packagePath: '/test/package',
+        },
+        executor,
+      );
+
+      expect(result).toEqual({
+        content: [
+          {
+            type: 'text',
+            text: "Error: Swift package build failed\nDetails: main.swift:10:25: error: cannot find type 'DOESNOTEXIST' in scope\nlet broken: DOESNOTEXIST = 42",
+          },
+        ],
+        isError: true,
+      });
+    });
+
     it('should handle spawn error', async () => {
       const executor = async () => {
         throw new Error('spawn ENOENT');

--- a/src/mcp/tools/swift-package/__tests__/swift_package_test.test.ts
+++ b/src/mcp/tools/swift-package/__tests__/swift_package_test.test.ts
@@ -218,6 +218,32 @@ describe('swift_package_test plugin', () => {
       });
     });
 
+    it('should include stdout diagnostics when stderr is empty on test failure', async () => {
+      const mockExecutor = createMockExecutor({
+        success: false,
+        error: '',
+        output:
+          "main.swift:10:25: error: cannot find type 'DOESNOTEXIST' in scope\nlet broken: DOESNOTEXIST = 42",
+      });
+
+      const result = await swift_package_testLogic(
+        {
+          packagePath: '/test/package',
+        },
+        mockExecutor,
+      );
+
+      expect(result).toEqual({
+        content: [
+          {
+            type: 'text',
+            text: "Error: Swift package tests failed\nDetails: main.swift:10:25: error: cannot find type 'DOESNOTEXIST' in scope\nlet broken: DOESNOTEXIST = 42",
+          },
+        ],
+        isError: true,
+      });
+    });
+
     it('should handle spawn error', async () => {
       const mockExecutor = async () => {
         throw new Error('spawn ENOENT');

--- a/src/mcp/tools/swift-package/swift_package_build.ts
+++ b/src/mcp/tools/swift-package/swift_package_build.ts
@@ -57,7 +57,7 @@ export async function swift_package_buildLogic(
   try {
     const result = await executor(['swift', ...swiftArgs], 'Swift Package Build', false, undefined);
     if (!result.success) {
-      const errorMessage = result.error ?? result.output ?? 'Unknown error';
+      const errorMessage = result.error || result.output || 'Unknown error';
       return createErrorResponse('Swift package build failed', errorMessage);
     }
 

--- a/src/mcp/tools/swift-package/swift_package_clean.ts
+++ b/src/mcp/tools/swift-package/swift_package_clean.ts
@@ -26,7 +26,7 @@ export async function swift_package_cleanLogic(
   try {
     const result = await executor(['swift', ...swiftArgs], 'Swift Package Clean', false, undefined);
     if (!result.success) {
-      const errorMessage = result.error ?? result.output ?? 'Unknown error';
+      const errorMessage = result.error || result.output || 'Unknown error';
       return createErrorResponse('Swift package clean failed', errorMessage);
     }
 

--- a/src/mcp/tools/swift-package/swift_package_test.ts
+++ b/src/mcp/tools/swift-package/swift_package_test.ts
@@ -67,7 +67,7 @@ export async function swift_package_testLogic(
   try {
     const result = await executor(['swift', ...swiftArgs], 'Swift Package Test', false, undefined);
     if (!result.success) {
-      const errorMessage = result.error ?? result.output ?? 'Unknown error';
+      const errorMessage = result.error || result.output || 'Unknown error';
       return createErrorResponse('Swift package tests failed', errorMessage);
     }
 

